core/capabilities/launcher.go

```diff
@@ -161,12 +182,21 @@ func (w *launcher) Launch(ctx context.Context, state *registrysyncer.LocalRegist
 	allPeers := make(map[ragetypes.PeerID]p2ptypes.StreamConfig)
 
 	publicDONs := []registrysyncer.DON{}
+
+	myDON := state.IDsToDONs[registrysyncer.DonID(
+		state.IDsToNodes[w.peerWrapper.GetPeer().ID()].WorkflowDONId,
+	)].DON
+
 	for _, id := range allDONIDs {
 		d := state.IDsToDONs[id]
 		if !d.DON.IsPublic {
 			continue
 		}
 
+		if filterDon2Don(myDON, d) {
+			continue
+		}
+
 		publicDONs = append(publicDONs, d)
```

Comment:
You don't want to filter out your own DON from here. Also technically one node can be a member of multiple DONs (see logic below).